docs: restart events doc edit (#7017)
strimzi/strimzi-kafka-operator: documentation/modules/deploying/ref-operator-restart-events-reasons.adoc

`ClusterCaCertKeyReplaced`:: The key used to sign the cluster's CA certificates has been replaced, and the pod is being restarted as part of the CA renewal process.
`ConfigChangeRequiresRestart`:: Some Kafka configuration properties are changed dynamically, but others require that the broker be restarted.
`CustomListenerCaCertChanged`:: The CA certificate used to secure the Kafka network listeners has changed, and the pod is restarted to use it.
`FileSystemResizeNeeded`:: The file system size has been increased, and a restart is needed to apply it.
`JbodVolumesChanged`:: A disk was added or removed from the Kafka volumes, and a restart is needed to apply the change.
`ManualRollingUpdate`:: A user annotated the pod, or the `StatefulSet`` or `StrimziPodSet` set it belongs to, to trigger a restart.
`PodForceRestartOnError`:: An error occurred that requires a pod restart to rectify.
`PodHasOldGeneration`:: The `StatefulSet` that the pod is a member of has been updated, so the pod needs to be recreated.
`PodHasOldRevision`:: The `StrimziPodSet` that the pod is a member of has been updated, so the pod needs to be recreated.

> `PodStuck`:: The pod is still pending, and is either unschedulable or not scheduled, so the operator has restarted the pod as a final attempt to get it running.

Reviewer: I don't think _unscheduable_ is not in the MW dictionary.
```suggestion
`PodStuck`:: The pod is still pending, and either is not scheduled or can not be scheduled, so the operator has restarted the pod in a final attempt to get it running.
```

> `PodUnresponsive`:: Strimzi was unable to connect to the pod, which can indicate a broker not starting properly, so the operator will restart it to try to resolve the issue.

Reviewer: ```suggestion
`PodUnresponsive`:: Strimzi was unable to connect to the pod, which can indicate a broker not starting correctly, so the operator will restart it in an attempt to resolve the issue.
```
